### ISSUE TITLE
[BUG] ML4W Welcome module in waybar is not working due to ags update to 2.0.1

### ISSUE THREAD

**Issue by Koushikg88** [mylinuxforwork/dotfiles#443]
**Describe the bug**
After updating the latest update the ml4w welcome module button is not woking.
no response if clicked the button.

**To Reproduce**
update to the latest version.
![screenshot_14112024_235508](https://github.com/user-attachments/assets/9c6917b4-3828-4ebe-8546-b5e0620effa4)

**Expected behavior**

**Distribution (please complete the following information):**
 - Arch Linux

**Comment by mylinuxforwork** [comment]
@Koushikg88 You can start the apps with
com.ml4w.welcome
com.ml4w.hyprland.settings
com.ml4w.dotfilessettings

Or from the rofi app starter.

**Comment by mgeldi** [comment]
The issue is related to the args package being updated recently to version 2. It is now [version 2.0.1](https://github.com/Aylur/ags/releases/tag/v2.0.1).
This removed the previous option to launch sidebar.
from the .config/waybar/modules.json:
// ML4W Welcome App
  "custom/ml4w-welcome": {
    "on-click": "ags -t sidebar",
    "on-click-right": "sleep 0.1 && ~/.config/ml4w/apps/ML4W_Welcome-x86_64.AppImage",
    "format": " ",
    "tooltip-format": "Open Sidebar"
  },

As you can see, it was previously: ags -t sidebar. Running this shows that it's no longer supported. Looks like the syntax should now be something like "ags run sidebar" instead of -t. Not sure about this though.
@mylinuxforwork

**Comment by mylinuxforwork** [comment]
@mgeldi @Koushikg88 Yes. Completely new code base. Will test the new start command. If it’s not working, I will replace the sidebar with the welcome app for now and add there two buttons for the settings and hyprland app.

**Comment by Koushikg88** [comment]
The clock button also not showing calander. If I click the button, no response is there.

**Comment by mylinuxforwork** [comment]
> The clock button also not showing calander. If I click the button, no response is there. Also the AI support button is also not working.

It is based on ags too.

**Comment by mylinuxforwork** [comment]
Some resources:

- https://github.com/Aylur/ags/releases/tag/v2.0.0
- https://aylur.github.io/ags/
- https://github.com/Aylur/ags/discussions/615
- https://github.com/kotontrion/PKGBUILDS/blob/main/agsv1/PKGBUILD

**Comment by keremenci** [comment]
Amazing. They didn't even bother to notify anyone of the changes. Their suggestion is for everyone to rewrite their configs from scratch.

**Comment by edzhabs** [comment]
what is the work around for ags? I added this files as protected that's its not updated on my end.

> The issue is related to the args package being updated recently to version 2. It is now [version 2.0.1](https://github.com/Aylur/ags/releases/tag/v2.0.1). This removed the previous option to launch sidebar. from the .config/waybar/modules.json: // ML4W Welcome App "custom/ml4w-welcome": { "on-click": "ags -t sidebar", "on-click-right": "sleep 0.1 && ~/.config/ml4w/apps/ML4W_Welcome-x86_64.AppImage", "format": " ", "tooltip-format": "Open Sidebar" },
> 
> As you can see, it was previously: ags -t sidebar. Running this shows that it's no longer supported. Looks like the syntax should now be something like "ags run sidebar" instead of -t. Not sure about this though. @mylinuxforwork

i tried changing to `ags run sidebar`, it's not working. 

> * [Guide to migrate ? Aylur/ags#615](https://github.com/Aylur/ags/discussions/615)

I also saw your comment and looks like you're currently working on it. Thank you @mylinuxforwork 

These commands are working though.
> com.ml4w.welcome
> com.ml4w.hyprland.settings
> com.ml4w.dotfilessettings

**Comment by gajama** [comment]
I took the pragmatic decision to, at least temporarily, rollback the version of AGS installed on my system to 1.8.2:

- Git clone the AUR repo for Aylur's GTK Shell: 
`git clone https://aur.archlinux.org/aylurs-gtk-shell.git && cd aylurs-gtk-shell`
- Checkout the last pre v2 commit:
`git checkout 2ab04a8ee`
- `makepkg`
- `sudo pacman -U aylurs-gtk-shell-1.8.2-3-x86_64.pkg.tar.lz4`

No doubt this could end up causing me other problems somewhere along the line, but it works for now and hopefully the upgrade, or a better temporary fix, will come along before that happens!

**Comment by mrjxtr** [comment]
Having the same issues ... no wonder when I try to `yay -S ags` it fails.

```bash
❯ yay -S ags
AUR Explicit (1): ags-3.6.1.30-1
AUR Dependency (1): sdl2_sound-2.0.2-1
Sync Make Dependency (1): cmake-3.31.0-1
:: PKGBUILD up to date, skipping download: ags
:: PKGBUILD up to date, skipping download: sdl2_sound
  2 sdl2_sound                       (Build Files Exist)
  1 ags                              (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> 
  2 ags                              (Build Files Exist)
  1 sdl2_sound                       (Build Files Exist)
==> Diffs to show?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> 
==> Making package: ags 3.6.1.30-1 (Mon 18 Nov 2024 09:34:56 PM +08)
==> Retrieving sources...
  -> Found ags-3.6.1.30.tar.gz
==> WARNING: Skipping verification of source file PGP signatures.
==> Validating source files with sha256sums...
    ags-3.6.1.30.tar.gz ... Passed
==> Making package: sdl2_sound 2.0.2-1 (Mon 18 Nov 2024 09:34:57 PM +08)
==> Retrieving sources...
  -> Found SDL2_sound-2.0.2.tar.gz
  -> Found SDL2_sound-2.0.2.tar.gz.sig
==> WARNING: Skipping verification of source file PGP signatures.
==> Validating source files with sha256sums...
    SDL2_sound-2.0.2.tar.gz ... Passed
    SDL2_sound-2.0.2.tar.gz.sig ... Skipped
:: Remove make dependencies after install? [y/N] y
:: (1/2) Parsing SRCINFO: ags
:: (2/2) Parsing SRCINFO: sdl2_sound
gpg: error reading key: No public key

 :: PGP keys need importing:
 -> 50F91FB550EC32AB4A710286FA148B892AB48044, required by: sdl2_sound
:: Import? [Y/n] y
:: Importing keys with gpg...
gpg: keyserver receive failed: No keyserver available
 -> problem importing keys
 ```

**Comment by Iss-in** [comment]
> Having the same issues ... no wonder when I try to `yay -S ags` it fails.
> 
> ```shell
> :: OK: rofi found.
> :: OK: dunst found.
> :: OK: waybar found.
> :: OK: hyprpaper found.
> :: OK: hyprlock found.
> :: OK: hypridle found.
> :: OK: hyprshade found.
> :: OK: wal found.
> :: OK: gum found.
> :: OK: wlogout found.
> :: ERROR: ags doesn't exists. Please install it with yay -S ags
> :: OK: magick found.
> :: OK: figlet found.
> :: OK: waypaper found.
> ```
> 
> ```shell
> ❯ yay -S ags
> AUR Explicit (1): ags-3.6.1.30-1
> AUR Dependency (1): sdl2_sound-2.0.2-1
> Sync Make Dependency (1): cmake-3.31.0-1
> :: PKGBUILD up to date, skipping download: ags
> :: PKGBUILD up to date, skipping download: sdl2_sound
>   2 sdl2_sound                       (Build Files Exist)
>   1 ags                              (Build Files Exist)
> ==> Packages to cleanBuild?
> ==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
> ==> 
>   2 ags                              (Build Files Exist)
>   1 sdl2_sound                       (Build Files Exist)
> ==> Diffs to show?
> ==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
> ==> 
> ==> Making package: ags 3.6.1.30-1 (Mon 18 Nov 2024 09:34:56 PM +08)
> ==> Retrieving sources...
>   -> Found ags-3.6.1.30.tar.gz
> ==> WARNING: Skipping verification of source file PGP signatures.
> ==> Validating source files with sha256sums...
>     ags-3.6.1.30.tar.gz ... Passed
> ==> Making package: sdl2_sound 2.0.2-1 (Mon 18 Nov 2024 09:34:57 PM +08)
> ==> Retrieving sources...
>   -> Found SDL2_sound-2.0.2.tar.gz
>   -> Found SDL2_sound-2.0.2.tar.gz.sig
> ==> WARNING: Skipping verification of source file PGP signatures.
> ==> Validating source files with sha256sums...
>     SDL2_sound-2.0.2.tar.gz ... Passed
>     SDL2_sound-2.0.2.tar.gz.sig ... Skipped
> :: Remove make dependencies after install? [y/N] y
> :: (1/2) Parsing SRCINFO: ags
> :: (2/2) Parsing SRCINFO: sdl2_sound
> gpg: error reading key: No public key
> 
>  :: PGP keys need importing:
>  -> 50F91FB550EC32AB4A710286FA148B892AB48044, required by: sdl2_sound
> :: Import? [Y/n] y
> :: Importing keys with gpg...
> gpg: keyserver receive failed: No keyserver available
>  -> problem importing keys
> ```

what it has got to do with this issue ? its your own os config problem

**Comment by gajama** [comment]
The AUR package is called `aylurs-gtk-shell`, presumably because the acronym `ags` was already taken by the Adventure Game Studio. So `yay -S aylurs-gtk-shell`.

**Comment by mrjxtr** [comment]
> The AUR package is called `aylurs-gtk-shell`, presumably because the acronym `ags` was already taken by the Adventure Game Studio. So `yay -S aylurs-gtk-shell`.

Awesome! did the trick! thanks

**Comment by qioxoip** [comment]
> I took the pragmatic decision to, at least temporarily, rollback the version of AGS installed on my system to 1.8.2:
> 
> * Git clone the AUR repo for Aylur's GTK Shell:
>   `git clone https://aur.archlinux.org/aylurs-gtk-shell.git && cd aylurs-gtk-shell`
> * Checkout the last pre v2 commit:
>   `git checkout 2ab04a8ee`
> * `makepkg`
> * `sudo pacman -U aylurs-gtk-shell-1.8.2-3-x86_64.pkg.tar.lz4`
> 
> No doubt this could end up causing me other problems somewhere along the line, but it works for now and hopefully the upgrade, or a better temporary fix, will come along before that happens!

This helped me also. I had to install typescript as a dependency but after that the downgrade did work. won´t be upgrading anzthing within the next few days though. ;-)

**Comment by mylinuxforwork** [comment]
Fixes with 2.9.6.7. Brings ags v2 support.

**Comment by mrjxtr** [comment]
> Fixes with 2.9.6.7. Brings ags v2 support.

I updated everything and it still does not work on my end... do I need to uninstall the `aylurs-gtk-shell` that I installed?

Why I installed it:

> The AUR package is called `aylurs-gtk-shell`, presumably because the acronym `ags` was already taken by the Adventure Game Studio. So `yay -S aylurs-gtk-shell`.

**Comment by gajama** [comment]
Did you roll back `aylurs-gtk-shell` to v1.8.x?

Updating ML4W should update `aylurs-gtk-shell` as well, unless you take steps to stop that happening. It did for me.

`aylurs-gtk-shell` is what the AUR package for `ags` is called. You need that. ML4W installs it as part of its own installation process. Maybe try reinstalling the current version of `aylurs-gtk-shell` if you did roll it back.
